test/modern_bpf/test_suites/syscall_enter_suite/connect_e.cpp

```diff
@@ -0,0 +1,255 @@
+#include "../../event_class/event_class.h"
+
+#if defined(__NR_connect)
+
+#include <netdb.h>
+#include <sys/un.h>
+
+TEST(SyscallEnter, connectE_INET)
+{
+	auto evt_test = new event_test(__NR_connect, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	struct sockaddr_in server_addr;
+	bzero(&server_addr, sizeof(server_addr));
+
+	server_addr.sin_family = AF_INET;
+	server_addr.sin_port = htons(IPV4_PORT_SERVER);
+
+	/// TODO: this is not a syscall probably we need to change the name of the helper
+	/// in something more generic...
+	assert_syscall_state(SYSCALL_SUCCESS, "inet_pton", inet_pton(AF_INET, IPV4_SERVER, &server_addr.sin_addr), NOT_EQUAL, -1);
+	/*  We pass an invalid socket fd so the `connect` must fail. */
+	int32_t mock_fd = 5;
```

Comment:
Not sure if `-1` would be better.. just in case fd 5 is valid (but might not be socket)

Reply:
reading further, passing `-1` have different semantics... maybe have a higher number but other than that, this is resolved.

Reply:
Probably i could remove this `if` since we can always catch this struct even if the syscall fails, it could be useful!

```c
	if(socket_fd > 0)
	{
		unsigned long sockaddr_ptr = extract__syscall_argument(regs, 1);
		u16 addrlen = (u16)extract__syscall_argument(regs, 2);
		auxmap__store_sockaddr_param(auxmap, sockaddr_ptr, addrlen);
	}
	else
	{
		auxmap__store_empty_param(auxmap);
	}
````

So i can put also `-1` in the test without problems

Reply:
+1 for being useful. If you other think to keep this `if`, then change to `socket_fd >= 0` because `stdout` could be closed or one could play around with `dup` calls.

Reply:
i've removed the `if` :)

Reply:
Thanks... the new commits look nice with that change.